ibtk/src/boundary/HierarchyGhostCellInterpolation.cpp

```diff
@@ -229,15 +217,15 @@ HierarchyGhostCellInterpolation::initializeOperatorState(
         }
     }
 
-    d_coarsen_strategy = NULL;
+    d_coarsen_strategy = nullptr;
 
     d_coarsen_scheds.resize(d_finest_ln + 1);
     if (registered_coarsen_op)
     {
         for (int src_ln = std::max(1, d_coarsest_ln); src_ln <= d_finest_ln; ++src_ln)
         {
-            Pointer<PatchLevel<NDIM> > level = d_hierarchy->getPatchLevel(src_ln);
-            Pointer<PatchLevel<NDIM> > coarser_level = d_hierarchy->getPatchLevel(src_ln - 1);
+            Pointer<PatchLevel<NDIM>> level = d_hierarchy->getPatchLevel(src_ln);
```

Comment:
I propose making this `auto`. I think that it should be clear that the result is a `PatchLevel`.

But, perhaps those kinds of changes can go into a follow-up PR.

Reply:
In general, I would suggest trying to convert `Pointer<T>` to `auto` wherever it makes sense, because it will be that much less code to change when moving from SAMRAI 2 to SAMRAI 3.